contracts/root/lib/ChildChainVerifier.sol

```diff
@@ -0,0 +1,167 @@
+pragma solidity ^0.5.2;
+
+import { RLPReader } from "solidity-rlp/contracts/RLPReader.sol";
+import { BytesLib } from "../../common/lib/BytesLib.sol";
+import { Common } from "../../common/lib/Common.sol";
+import { RLPEncode } from "../../common/lib/RLPEncode.sol";
+import { Common } from "../../common/lib/Common.sol";
+import { MerklePatriciaProof } from "../../common/lib/MerklePatriciaProof.sol";
+
+import { Registry } from "../../common/Registry.sol";
+
+library ChildChainVerifier {
+  using RLPReader for bytes;
+  using RLPReader for RLPReader.RLPItem;
+
+  function processBurnReceipt(
+    bytes memory receiptBytes, bytes memory path, bytes memory receiptProof,
+    bytes32 receiptRoot, address sender)
+    internal
+    pure
+    returns (address rootToken, address childToken, uint256 amountOrTokenId)
+  {
+    RLPReader.RLPItem[] memory items = receiptBytes.toRlpItem().toList();
+    require(items.length == 4, "MALFORMED_RECEIPT");
+    // Do any other fields other than items[3] need to be checked?
```

Comment:
nope mostly we care about Burnt/withdraw receipt and its inclusion, in future, we can think of some mechanism to reduce all these RLP bytes Data.